### PR TITLE
Fix blocks missing translations in simple sites

### PR DESCRIPTION
**Pull request by monsieur-z**
Fixes p1694788078746839-slack-CDLH4C1UZ

## Proposed changes:

https://github.com/Automattic/jetpack/pull/32698, https://github.com/Automattic/jetpack/pull/32815, and https://github.com/Automattic/jetpack/pull/33073 introduced localization issues for simple sites: block titles and descriptions were displayed in English no matter the user's selected language. This PR fixes it.

The blocks involved are: Business Hours, AI Chat, Amazon, Blogroll, Google Docs Embed, Recipe, Create with Voice.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Review the code
- Regression testing:
  - Spin up a test site with this PR. This could be a self-hosted site or a simple site.
  - Create a new post.
  - Notice that the aforementioned blocks are not broken.
- Localization testing:
  - Download this branch.
  - Build the plugin: `jetpack build plugins/jetpack`.
  - Remote sync the plugin with your sandbox (see PCYsg-eg0-p2).
  - Change the language of your WordPress.com account to something different than English.
  - In WordPress.com, select the site linked to your sandbox and create a new post.
  - In the block inserter, notice that the block titles and descriptions for the blocks above are translated.

_Blocks info translated to French_
Screenshot 2023-09-19 at 9 47 38 AM
Screenshot 2023-09-19 at 9 47 49 AM
Screenshot 2023-09-19 at 9 47 57 AM
Screenshot 2023-09-19 at 9 48 04 AM
Screenshot 2023-09-19 at 9 48 15 AM
Screenshot 2023-09-19 at 9 48 23 AM
Screenshot 2023-09-19 at 9 48 31 AM

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-chat/editor.js [hunk 1]
```diff
@@ -7,8 +7,6 @@ import './editor.scss';
 import './components/feedback/style.scss';
 
 registerJetpackBlockFromMetadata( metadata, {
-	// The API version needs to be explicitly specified in this instance for styles to be loaded.
-	apiVersion: metadata.apiVersion,
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/ai-chat/editor.js:11]: Metadata are added to settings in `registerJetpackBlockFromMetadata` anyway.

File: projects/plugins/jetpack/extensions/blocks/business-hours/block.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
 	"$schema": "https://schemas.wp.org/trunk/block.json",
-	"apiVersion": 3,
+	"apiVersion": 1,
```
Comment (monsieur-z) [inline projects/plugins/jetpack/extensions/blocks/business-hours/block.json:3]: With the changes brought to `registerJetpackBlockFromMetadata `, the implementation of the `edit` function (I believe) requires the block to use the first version of the API.

**Review by andrii-lysenko — APPROVED** [review]
LGTM!

I think this can be merged after issues with the builds are resolved.

File: projects/plugins/jetpack/extensions/shared/register-jetpack-block.js [hunk 3]
```diff
@@ -59,6 +62,24 @@ export default function registerJetpackBlock( name, settings, childBlocks = [],
...
+// when it is passed the metadata object as the first param. I haven't found this to be true
+// with the latest version of `@wordpress/blocks` (12.18.0) at the time of writing this.
+// https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#javascript
+const localizeBlockProperties = ( metadata, props ) => {
+	const result = { ...props };
+
+	for ( const prop in props ) {
+		if ( LOCALIZED_BLOCK_PROPERTIES.includes( prop ) ) {
+			// eslint-disable-next-line
+			result[ prop ] = _x( props[ prop ], null, metadata.textdomain || DEFAULT_TEXTDOMAIN );
```
Comment (anomiex) [inline projects/plugins/jetpack/extensions/shared/register-jetpack-block.js:76]: (most of this is already known to the people involved via Slack discussions, but to close the loop I'll repeat it here)

I think this only accidentally works if your translation files still have the translations from https://github.com/Automattic/jetpack/blob/427e370fe89c3ed1944a1ba93e97be94f00fd034/projects/plugins/jetpack/extensions/blocks/business-hours/index.js#L76-L77 that were removed in #32698.

WordPress intends for the translation to happen on the PHP side during block registration ([here](https://github.com/WordPress/WordPress/blob/f6307ff5fcfde0081cf94194c56defb8370b8c94/wp-includes/blocks.php#L443-L450)), then the already-translated texts are sent to JS via a generated call to `wp.blocks.unstable__bootstrapServerSideBlockDefinitions` ([here](https://github.com/WordPress/wordpress-develop/blob/9db15f2863e3625451d929767253bb1bb6c97388/src/wp-admin/site-editor.php#L129-L132), [here](https://github.com/WordPress/wordpress-develop/blob/9db15f2863e3625451d929767253bb1bb6c97388/src/wp-admin/edit-form-blocks.php#L108-L111), [here](https://github.com/WordPress/wordpress-develop/blob/9db15f2863e3625451d929767253bb1bb6c97388/src/wp-admin/widgets-form-blocks.php#L49-L52), or [here](https://github.com/WordPress/wordpress-develop/blob/9db15f2863e3625451d929767253bb1bb6c97388/src/wp-includes/class-wp-customize-widgets.php#L864-L867)).

That PHP code path is only reached when using `register_block_type_from_metadata()`, or passing a path to the block.json to `register_block_type()`. We instead [pass a WP_Block_Type object](https://github.com/Automattic/jetpack/blob/57c470f7a4ad3e87fb76d4f75df86d442c414ee2/projects/packages/blocks/src/class-blocks.php#L53-L65). So we'd either need to change our code to pass the block.json path, or duplicate the relevant logic (plus any other needed logic for any other core block.json registeration features) when creating our `WP_Block_Type`.

**Comment by monsieur-z** [comment]
Close in favour of #33326.

**Pull request closed without merging**